pkg/network/usm/kafka_monitor_test.go

```diff
@@ -1035,6 +1038,32 @@ func testKafkaFetchRaw(t *testing.T, tls bool) {
 			},
 			numFetchedRecords: 3,
 		},
+		{
+			name: "with error code",
+			buildResponse: func() kmsg.FetchResponse {
+				record := makeRecord()
+				var records []kmsg.Record
+				for i := 0; i < 5; i++ {
+					records = append(records, record)
+				}
+
+				recordBatch := makeRecordBatch(records...)
+				var batches []kmsg.RecordBatch
+				for i := 0; i < 4; i++ {
+					batches = append(batches, recordBatch)
+				}
+
+				partition := makeFetchResponseTopicPartition(3, batches...)
+				var partitions []kmsg.FetchResponseTopicPartition
+				for i := 0; i < 3; i++ {
+					partitions = append(partitions, partition)
+				}
+
+				return makeFetchResponse(makeFetchResponseTopic(topic, partitions...))
+			},
+			numFetchedRecords: 5 * 4 * 3,
+			errorCode:         3,
```

Comment:
Maybe we should have a test case which includes a fetch response with partitions both with and without errors?

Reply:
Will do